### ISSUE TITLE
Code 39 not working with dash

### ISSUE THREAD

**Issue by dhruvJaiswal** [barnhill/barcodelib#147]
Barcode is getting generated using Code 39 from barcodelib but it is not getting scanned when using dash. all other barcode without dash is getting scanned

**Comment by rob313663** [comment]
Please attach image of the barcode and the contents it should contain
together with the code used to generate it.

You have probably used a character that is not encodable in Code 39.

**Comment by barnhill** [comment]
Just tested this on the upcoming branch for cross-platform compatibility and this scans just fine with Code39 and a dash added.  I have to think that @rob313663 is correct and its not a standard dash character.

![image](https://github.com/barnhill/barcodelib/assets/3878158/345df071-58c7-468d-99a7-3f575589f562)

**Comment by barnhill** [comment]
I think it has to do with the data formatted to be encoded.  I tried your input with another online barcode generator and then scanned it on the same website and its detecting the same incorrect input data.

Used to generate:
https://www.barcodesinc.com/generator/index.php
![image](https://github.com/barnhill/barcodelib/assets/3878158/630a98a8-ca18-49f8-a0a1-974fd63d2bc5)


Used to read:
https://online-barcode-reader.inliteresearch.com/
![image](https://github.com/barnhill/barcodelib/assets/3878158/44e56a25-203b-4145-94e6-14eb19776042)

**Comment by barnhill** [comment]
Code39 and all other types should probably contain data validation and return error codes when invalid chars are detected

**Comment by rob313663** [comment]
Code 39 has supported dash since it was invented in 1974.

**Comment by barnhill** [comment]
Yep and it's supported in this lib too as the evidence above shows.  My last comment was regarding invalid chars.
